packages/botonic-react/index.d.ts

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import * as core from '@botonic/core';
+
+export interface MessageProps{
+  type? : string;
+  from? : string;
+  delay? : number;
+  typing? : number;
+  children : any;
+  json: any;
+  style: any;
+}
+export class Message extends React.Component<MessageProps, any> {}
+
+
+//TODO inherit from MessageProps?
+export interface TextProps{
+  delay?: number;
+  typing?: number;
+}
+export class Text extends React.Component<TextProps, any> {}
+
+export interface ButtonProps{
+  payload?: string;
+  url?: string;
+}
+export class Button extends React.Component<ButtonProps, any> {}
+
+export interface ButtonProps{
+  payload?: string;
+  url?: string;
+}
+export class Reply extends React.Component<ButtonProps, any> {}
+
+export interface PicProps{
+  src: string;
+}
+export class Pic extends React.Component<PicProps, any> {}
+export class Image extends React.Component<PicProps, any> {}
+
+export class Carousel extends React.Component<any, any> {}
+export class Title extends React.Component<any, any> {}
+export class Subtitle extends React.Component<any, any> {}
+export class Element extends React.Component<any, any> {}
+
+export type Locales = {[id: string]: string|string[]|Locales };
+
+export class App {
+  routes: core.Route[]
+  locales: Locales;
+  integrations?: {[id: string]: any};
+  theme?: string;
+  plugins: { [id: string]: any};
+  defaultTyping : number;
+  defaultDelay : number;
+
+  constructor(app: {
+    routes: core.Route[],
+    locales: Locales,
+    integrations?: {[id: string]: any};
+    theme?: string,
+    plugins?: { [id: string]: any},
+    defaultTyping?: number,
+    defaultDelay?: number
+  });
+}
+
+export interface Input {
+  type: string; // text, postback...
+  payload?: string;
+  data?: string;
+}
+
+export interface Session {
+  is_first_interaction?: boolean,
+  last_session?: any,
+  user: {
+    id: string
+    // login
+    username?: string,
+    // person name
+    name?: string,
+    // whatsapp, telegram,...
+    provider?: string,
+    // The provider's user id
+    provider_id?: string,
+    extra_data?: any
+  },
+  bot: {
+    id: string,
+    name?: string
+  }
+  __locale?: string,
+  __retries?: number
+}
```

Comment:
Maybe we can leave the session interface as just an object (any), so we're not rigged to the current data model. This is something that can change a lot and also, everytime the user wants to add properties (which will be very common), he'll go into "casting hell"...

Reply:
yes, but it's very useful to have at least the types of the most stable ones. Developers using JS will not be affected, and developers using TS are used casting to any or disabling the warnings (typically via IDE hints)